### PR TITLE
Added the ability to get recipient tabs as an optional parameter

### PR DESCRIPTION
**Pull request by mikebz**

### CODE REVIEW

File: jsconfig.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "compilerOptions": {
+        "target": "ES6",
+        "module": "commonjs"
+    }
+}
```
Comment (Whoaa512) [inline jsconfig.json]: What's the purpose of this file?

Comment (mikebz) [inline jsconfig.json]: This is for the VS Code editor instructing it to use ES6 syntax.

On Oct 27, 2015, at 9:51 AM, C.J. Winslow <notifications@github.com<mailto:notifications@github.com>> wrote:


In jsconfig.json<https://github.com/docusign/DocuSign-Node-Client/pull/36#discussion_r43150218>:

> @@ -0,0 +1,6 @@
> +{
> +    "compilerOptions": {
> +        "target": "ES6",
> +        "module": "commonjs"
> +    }
> +}


What's the purpose of this file?

-

File: components/envelopes.js [hunk 2]
```diff
@@ -847,14 +855,14 @@ function getTemplateView (apiToken, baseUrl, templateId, returnUrl) {
  * @param {string} apiToken - DocuSign API OAuth2 access token.
  * @param {string} baseUrl - DocuSign API base URL.
  * @param {string} envelopeId - ID of envelope to get list of recipients from.
+ * @param {boolean} [include_tabs] - When true the tab information associated with the recipient is included in the response.
  * @returns {Promise} - A thenable bluebird Promise fulfilled with the list of recipients
  */
-function getRecipients (apiToken, baseUrl, envelopeId) {
+function getRecipients (apiToken, baseUrl, envelopeId, include_tabs) {
```
Comment (Whoaa512) [inline components/envelopes.js]: You should provide a default value for `include_tabs` like so:
```javascript
include_tabs = include_tabs != null ? include_tabs : false;
```

File: test/component-tests/envelopes.spec.js [hunk 3]
```diff
@@ -134,6 +134,23 @@ test(function getRecipients (t) {
   });
 });
 
+test(function getRecipientsWithTabs (t) {
+  return client.envelopes.getRecipients(envelopeId, true)
```
Comment (Whoaa512) [inline test/component-tests/envelopes.spec.js]: In an effort to make our tests self-documenting, please define a variable describing what the `true` value represents here.
```javascript
let include_tabs = true;
```

File: components/envelopes.js [hunk 4]
```diff
@@ -217,11 +217,19 @@ exports.init = function (accountId, baseUrl, accessToken) {
...
      * @param {function} [callback] - Returns the list of recipients in the form of function(error, response).
      * @returns {Promise} - A thenable bluebird Promise; if callback is given it is called before the promise is resolved
      */
-    getRecipients: function (envelopeId, callback) {
-      return getRecipients(accessToken, baseUrl, envelopeId).asCallback(callback);
+    getRecipients: function (envelopeId, include_tabs, callback) {
+      // handle the case where people omit the tabs
+      if (arguments.length === 2 && Object.prototype.toString.call(include_tabs) === '[object Function]') {
+        callback = include_tabs;
+        include_tabs = false;
```
Comment (Whoaa512) [inline components/envelopes.js]: Please add a test to account for the logic handling when `include_tabs` is omitted.

File: test/component-tests/envelopes.spec.js (hunk 3 above)
Comment (mikebz) [inline test/component-tests/envelopes.spec.js]: good suggestion.  Fixing.

File: components/envelopes.js (hunk 2 above)
Comment (mikebz) [inline components/envelopes.js]: good suggestion.  fixing.

File: components/envelopes.js (hunk 4 above)
Comment (mikebz) [inline components/envelopes.js]: I think that is the test on line 139 of envelope.spec.js.  

  return client.envelopes.getRecipients(envelopeId, include_tabs)...

Let me know if you think there should be another test there.

Comment (Whoaa512) [inline components/envelopes.js]: Ah yea I forgot that the tests don't cover the callback cases. That's fine, we can just add an `/* istanbul ignore if */` here to appease the coverage gods.

**Comment by Whoaa512** [comment]
LGTM :+1:

**Pull request merged**